README.md

```diff
@@ -12,6 +12,93 @@ With its huge context window, we are able to feed the entire conversational hist
+              └───────────┘                  └─────┬─────┘
+                                                  │
+                                            ┌─────┴─────┐
+                                            │ Supabase  │
+                                            │ (Vector   │
+                                            │   DB)     │
+                                            └───────────┘
+```
+
+### Key Components
+
+#### Adapters (`src/adapters/`)
+- **OpenAI Adapter**: Handles interactions with OpenAI's GPT models for generating responses
+- **Voyage Adapter**: Generates embeddings using the voyage-large-2-instruct model
+- **Supabase Adapter**: Manages vector similarity search and storage using pgvector
+
+#### Vector Search System
+- Uses Voyage AI to generate 1024-dimensional embeddings
+- Stores embeddings in Supabase using pgvector extension
+- Combines vector similarity with text similarity for optimal matching
+- Implements efficient similarity search functions for both issues and comments
+
+#### Handlers (`src/handlers/`)
+- **Comment Handlers**: Process incoming GitHub webhook events
+- **Ground Truth System**: Validates LLM responses against known correct answers
+- **Ask LLM Handler**: Core logic for formulating queries and processing responses
+
+#### Helpers (`src/helpers/`)
+- **Token Utils**: Manages token counting and context window optimization
+- **Issue/PR Fetching**: Recursively retrieves related GitHub conversations
+- **Chat History Formatting**: Structures conversation history for LLM context
+- **Callback Proxy**: Manages asynchronous webhook callbacks
+
+### External Service Integration
+
+#### GitHub Integration
+- Processes webhook events for new comments
+- Recursively fetches conversation history from referenced issues/PRs
+- Supports both issues and pull request contexts
+
+#### OpenAI Integration
+- Uses GPT-4o model for generating responses
+- Optimizes context window usage for maximum relevance
+- Supports configurable model settings via plugin configuration
+
+#### Voyage AI Integration
+- Generates high-quality embeddings using voyage-large-2-instruct model
+- 1024-dimensional vectors for precise similarity matching
+- Optimized for technical and conversational content
+
+#### Supabase Vector Database
+- Uses pgvector extension for efficient vector similarity search
+- Combines vector similarity with text-based search for better results
+- Implements specialized functions for finding similar issues and comments
+- Stores and indexes embeddings for fast retrieval
+
+### Testing Infrastructure
+
+The project uses Jest for testing with comprehensive mocks for:
+- Database operations
+- GitHub API responses
+- Webhook handlers
+- Issue and repository templates
+
+Tests are organized to validate:
+- Core business logic
+- External service integrations
+- Webhook processing
+- Ground truth validation
+
 ## Installation
 
 `ubiquibot-config.yml`:
```

Comment:
All "ubiquibot" terminology has been deprecated in favor of "UbiquityOS" or "ubiquity-os" in lowercase slug form. 

`ubiquity-os.config.yml`